prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -0,0 +1,171 @@
+from powersimdata.utility.distance import haversine
+
+from prereise.gather.griddata.hifld.data_access import load
+
+
+def check_for_location_conflicts(substations):
+    """Check for multiple substations with identical lat/lon.
+
+    :param pandas.DataFrame substations: data frame of substations.
+    :raises ValueError: if multiple substations with identical lat/lon.
+    """
+    num_substations = len(substations)
+    num_lat_lon_groups = len(substations.groupby(["LATITUDE", "LONGITUDE"]))
+    if num_lat_lon_groups != num_substations:
+        num_collisions = num_substations - num_lat_lon_groups
+        raise ValueError(
+            f"There are {num_collisions} substations with duplicate lat/lon values"
+        )
+
+
+def get_substations_with_nonzero_lines(substations):
+    """Filter substations with LINES attribute equal to zero, and report the number
+    dropped.
+
+    :param pandas.DataFrame substations: data frame of all substations.
+    :return: (*pandas.DataFrame*) -- substations with non-zero values for LINES.
+    """
+    num_substations = len(substations)
+    num_substations_without_lines = len(substations.query("LINES == 0"))
+    print(
+        f"dropping {num_substations_without_lines} substations "
+        f"of {num_substations} total due to LINES parameter equal to 0"
+    )
+
+    return substations.query("LINES != 0").copy()
```

Comment:
According to the [documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.query.html), I think `query` will return a copy of filtered data frame, `copy` is redundant here.

Reply:
I think we do need this, if only to prevent unexpected warning messages. If we try to set the value on a data frame created via `query`, we get a `SettingWithCopyWarning`.
```python
>>> import pandas as pd
>>> df = pd.DataFrame({"foo": ["a", "b"], "bar": [1, 2]})
>>> df
  foo  bar
0   a    1
1   b    2
>>> q = df.query("foo == 'a'")
>>> q
  foo  bar
0   a    1
>>> q["bar"] = 3
<stdin>:1: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
>>> q
  foo  bar
0   a    3
>>> df
  foo  bar
0   a    1
1   b    2
>>> q.loc[0, "bar"] = 4
C:\Python39\lib\site-packages\pandas\core\indexing.py:1817: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)
>>> q
  foo  bar
0   a    4
>>> df
  foo  bar
0   a    1
1   b    2
>>> q2 = df.query("foo == 'a'").copy()
>>> q2["bar"] = 5
>>> q2.loc[0, "bar"] = 6
>>>
```

Reply:
Fair enough. Although I don't quite understand why it throws this warning even we set values on a copy of the get return, which is guaranteed by `query` unless `inplace` is set.

Reply:
I think if `inplace=True`, then the return is `None` and the original data frame is modified; otherwise, we _may_ get a 'view' of the DataFrame, rather than a copy. I guess either one will be a DataFrame object, but there are differences in some of the underlying attributes?
https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

Reply:
My references are here:
1. pandas documentation, where it says explicitly "a copy" under the inplace parameter doc strings. 
![image](https://user-images.githubusercontent.com/52716585/129403364-fe05c861-1d16-4674-af0e-daa8ddd8253e.png)
2. The accepted answer of this post: https://stackoverflow.com/questions/23296282/what-rules-does-pandas-use-to-generate-a-view-vs-a-copy

Reply:
I'll admit, I don't fully understand the internal pandas logic, and I'm poking at a semi-opaque box via toy examples. Maybe there's a difference between a copy of a slice of a dataframe vs. a copy of a dataframe?

Reply:
I don't know either ;) always learning along the way. I'm good with taking an explicit copy regardlessly.